jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassURLHelperImpl.java

```diff
@@ -47,12 +52,28 @@
 		initializeShareableClassloader(loader);
 	}
 	
-	private static native void init(); 
+	private static native void init();
+	
+	/* The passed-in URL path contains a jar file */
+	private boolean newJarFileCheck(URL convertedjarPath) {
+		boolean newJarFile = false;
+		String protocol = convertedjarPath.getProtocol();
+		String jarFile = convertedjarPath.getFile();
+		if ((protocol.equalsIgnoreCase("file") || protocol.equalsIgnoreCase("jar"))  //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
If the protocol is "jar", the file suffix doesn't matter. Although looking at recursiveJarTrim() the protocol will never be "jar" in the convertedjarPath.

Reply:
Already modified the code against the native function `getCpeTypeForProtocol`

Reply:
Already verified the failing test:
```
Testing: HelperCompatibilityTest 9
Test start time: 2019/07/04 11:45:20 Pacific Standard Time
Running command: "/.../j2sdk-image/jre/bin/java" -Xcompressedrefs -Xcompressedrefs -Xjit -Xgcpolicy:gencon  -Xshareclasses:name=HelperCompatibilityTests -Xbootclasspath/a:./Utils:. URLClassPathHelperURLHelperStaleEntryCompatibilityTest -testfile "/../openj9/test/TestConfig/scripts/testKitGen/../../../../jvmtest/functional/cmdLineTests/shareClassTests/SCHelperCompatibilityTests/props_unix/Test8.props" -javacdir /../j2sdk-image/bin
Time spent starting: 2 milliseconds
Time spent executing: 3299 milliseconds
Test result: PASSED
```